packages/client/src/views/RecordAudit/ActionMenu.tsx

```diff
@@ -0,0 +1,646 @@
+      (scope as any as string[]).includes('validate'))
+
+  return (
+    isCertified &&
+    userHasIssueScope && (
+      <DropdownMenu.Item
+        onClick={() => {
+          dispatch(clearCorrectionAndPrintChanges(declarationId as string))
+          dispatch(goToIssueCertificate(declarationId as string))
+        }}
+        disabled={!isDownloaded}
+      >
+        <Icon name="Handshake" color="currentColor" size="large" />
+        {intl.formatMessage(messages.issueCertificate)}
+      </DropdownMenu.Item>
+    )
+  )
+}
+
+const DeleteAction: React.FC<{
+  intl: IntlShape
+  handleDelete: () => void
+  declarationStatus?: string
+}> = ({ intl, handleDelete, declarationStatus }) =>
+  declarationStatus === SUBMISSION_STATUS.DRAFT && (
+    <DropdownMenu.Item onClick={handleDelete}>
+      <Icon name="Trash" color="currentColor" size="large" />
+      {intl.formatMessage(buttonMessages.deleteDeclaration)}
+    </DropdownMenu.Item>
+  )
+const UnassignAction: React.FC<{
+  intl: IntlShape
+  handleUnassign: () => void
+  isDownloaded: boolean
+  assignment?: GQLAssignmentData
+  scope: Scope
+}> = ({ intl, handleUnassign, isDownloaded, assignment, scope }) => {
+  const isAssignedToSomeoneElse = !isDownloaded && assignment
+
+  // @ToDo use: appropriate scope after configurable role pr is merged
+  const userHasUnassignScope =
+    scope && (scope as any as string[]).includes('register')
+
+  return (
+    ((isAssignedToSomeoneElse && userHasUnassignScope) || isDownloaded) && (
+      <DropdownMenu.Item onClick={handleUnassign}>
+        <Icon name="ArrowCircleDown" color="currentColor" size="large" />
+        {intl.formatMessage(buttonMessages.unassign)}
+      </DropdownMenu.Item>
+    )
+  )
+}
+
+const UnassignModal: React.FC<{
+  intl: IntlShape
+  close: (result: boolean | null) => void
+  assignment?: GQLAssignmentData
+  isDownloaded: boolean
+}> = ({ intl, close, assignment, isDownloaded }) =>
+  assignment && (
+    <ResponsiveModal
+      autoHeight
+      responsive={false}
+      title={intl.formatMessage(conflictsMessages.unassignTitle)}
+      actions={[
+        <TertiaryButton
+          id="cancel_unassign"
+          key="cancel_unassign"
+          onClick={() => {
+            close(null)
+          }}
+        >
+          {intl.formatMessage(buttonMessages.cancel)}
+        </TertiaryButton>,
+        <DangerButton
+          key="confirm_unassign"
+          id="confirm_unassign"
+          onClick={() => {
+            close(true)
+          }}
+        >
+          {intl.formatMessage(buttonMessages.unassign)}
+        </DangerButton>
+      ]}
+      show={true}
+      handleClose={() => close(null)}
+    >
+      {isDownloaded
+        ? intl.formatMessage(conflictsMessages.selfUnassignDesc)
+        : intl.formatMessage(conflictsMessages.regUnassignDesc, {
+            name: assignment.firstName + ' ' + assignment.lastName,
+            officeName: assignment.officeName
+          })}
```

Comment:
We could avoid having to pass an `assignment` object here by passing this component a `userName` and `officeName` for example. React cannot diff between objects so it helps with unnecessary re-renders.